go/tasks/plugins/webapi/athena/utils.go

```diff
@@ -0,0 +1,143 @@
+package athena
+
+import (
+	"context"
+
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core/template"
+
+	"github.com/lyft/flyteplugins/go/tasks/errors"
+
+	pluginsIdl "github.com/lyft/flyteidl/gen/pb-go/flyteidl/plugins"
+	"github.com/lyft/flytestdlib/utils"
+
+	pb "github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/ioutils"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/webapi"
+	"github.com/lyft/flytestdlib/logger"
+)
+
+func writeOutput(ctx context.Context, tCtx webapi.StatusContext, externalLocation string) error {
+	taskTemplate, err := tCtx.TaskReader().Read(ctx)
+	if err != nil {
+		return err
+	}
+
+	if taskTemplate.Interface == nil || taskTemplate.Interface.Outputs == nil || taskTemplate.Interface.Outputs.Variables == nil {
+		logger.Infof(ctx, "The task declares no outputs. Skipping writing the outputs.")
+		return nil
+	}
+
+	resultsSchema, exists := taskTemplate.Interface.Outputs.Variables["results"]
+	if !exists {
+		logger.Infof(ctx, "The task declares no outputs. Skipping writing the outputs.")
+		return nil
+	}
+
+	return tCtx.OutputWriter().Put(ctx, ioutils.NewInMemoryOutputReader(
+		&pb.LiteralMap{
+			Literals: map[string]*pb.Literal{
+				"results": {
+					Value: &pb.Literal_Scalar{
+						Scalar: &pb.Scalar{Value: &pb.Scalar_Schema{
+							Schema: &pb.Schema{
+								Uri:  externalLocation,
+								Type: resultsSchema.GetType().GetSchema(),
+							},
+						},
+						},
+					},
+				},
+			},
+		}, nil))
+}
+
+type QueryInfo struct {
+	QueryString string
+	Workgroup   string
+	Catalog     string
+	Database    string
+}
+
+func validateHiveQuery(hiveQuery pluginsIdl.QuboleHiveJob) error {
+	if hiveQuery.Query == nil {
+		return errors.Errorf(errors.BadTaskSpecification, "Query is a required field.")
+	}
+
+	if len(hiveQuery.Query.Query) == 0 {
+		return errors.Errorf(errors.BadTaskSpecification, "Query statement is a required field.")
+	}
+
+	return nil
+}
+
+func validatePrestoQuery(prestoQuery pluginsIdl.PrestoQuery) error {
+	if len(prestoQuery.Statement) == 0 {
+		return errors.Errorf(errors.BadTaskSpecification, "Statement is a required field.")
+	}
+
+	return nil
+}
+
+func extractQueryInfo(ctx context.Context, tCtx webapi.TaskExecutionContextReader) (QueryInfo, error) {
```

Comment:
do we want to do it this way, of is this just to get going with athena and we will add a proper plugin?

Reply:
Depending on how we land on the SQL Abstraction, we might have a more generic Sql Plugin with engine (and maybe engine-specific properties)... or continue to use Presto here... Athena is based on Presto and should be able to execute presto queries just fine...

Reply:
ok, btw where are we landing on that? I would recommend looking at getdbt for inspiration